fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -3048,17 +3053,56 @@ private boolean isEnableFillDataCacheImpl(Partition partition) throws AnalysisEx
                     dataCacheRange = Range.openClosed(PartitionKey.ofDateTime(lower), PartitionKey.ofDateTime(upper));
                     return partitionRange.isConnected(dataCacheRange);
                 } catch (Exception e) {
-                    LOG.warn("Table name: {}, datacache.partiton_duration: {}, Failed to compare with partition range. " +
-                            "Error: {}.", super.name, cacheDuration.toString(), e.getMessage());
+                    LOG.warn("Table name: {}, Partition name: {}, Datacache.partiton_duration: {}, Failed to check the " +
+                            " validaity of range partition. Error: {}.", super.name, partition.getName(),
+                            cacheDuration.toString(), e.getMessage());
                     return false;
                 }
-            } else {
-                // If the table was not partitioned by DATE/DATETIME, ignore the property "datacache.partition_duration" and
-                // enable data cache by default.
+            }
+        } else if (getPartitionInfo().isListPartition()) {
+            ListPartitionInfo listPartitionInfo = (ListPartitionInfo) getPartitionInfo();
+            List<Column> columns = listPartitionInfo.getPartitionColumns();
+            int dateTypeColumnIdx = IntStream.range(0, columns.size())
+                    .filter(i -> columns.get(i).getPrimitiveType().isDateType())
+                    .findFirst()
+                    .orElse(-1);
```

Comment:
Nit: Maybe you can use `org.apache.commons.collections4.ListUtils.indexOf` to simplify this statement.